### PR TITLE
Feature/cbl 2.8.x-- db upgrade feature

### PR DESCRIPTION
**Pull request by mgcouchbase**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

### CODE REVIEW

File: testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py [hunk 1]
```diff
@@ -352,3 +355,69 @@ def _upgrade_db(args):
...
+    testserver2.download()
+
+    log_info("Installing TestServer ...")
+    # Install TestServer app
+    if device_enabled:
+        testserver2.install_device()
+    else:
+        testserver2.install()
+
+    if device_enabled:
```
Comment (euniceatcouchbase) [inline testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py:403]: you can combine this if with the one above

**Review by euniceatcouchbase — APPROVED** [review]
with a minor request

File: testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py (hunk 1 above)
Comment (mgcouchbase) [reply testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py]: done

File: testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py [hunk 2]
```diff
@@ -352,3 +355,69 @@ def _upgrade_db(args):
...
+    db = Database(base_url)
+    source_db = db.create(cbl_db)
+    upgraded_cbl_doc_count = db.getCount(source_db)
+    cbl_doc_ids = db.getDocIds(source_db)
+    cbl_db_docs = db.getDocuments(source_db, cbl_doc_ids)
+    assert upgraded_cbl_doc_count == base_cbl_doc_count, " Number of docs count is not matching"
+    # Create few more docs and make sure the after upgrade we still create docs
+    db.create_bulk_docs(2, "cbl-upgrade-docs_new", db=source_db, channels=channels_sg)
+    upgraded_cbl_doc_count = db.getCount(source_db)
+    assert upgraded_cbl_doc_count > base_cbl_doc_count, " Number of docs count is not matching"
```
Comment (sridevi-15) [inline testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py:418]: As you know how many docs you are adding after the upgrade, It is better to assert with exact number as number of docs u r adding is static

File: testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py [hunk 3]
```diff
@@ -352,3 +355,69 @@ def _upgrade_db(args):
...
+    log_info("*" * 20)
+    log_info(cbl_db_docs)
```
Comment (sridevi-15) [inline testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py:420]: it is unnecessarily add too much logging . Can you remove this if you add it for debugging purpose for both log_info?

File: testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py [hunk 4]
```diff
@@ -352,3 +355,69 @@ def _upgrade_db(args):
...
+    debug_mode = params_from_base_suite_setup["debug_mode"]
+    device_enabled = params_from_base_suite_setup["device_enabled"]
+    community_enabled = params_from_base_suite_setup["community_enabled"]
+    testserver = params_from_base_suite_setup["testserver"]
+    test_name_cp = params_from_base_suite_setup["test_name_cp"]
+    channels_sg = ["ABC"]
+    # Create CBL database
+    db.create_bulk_docs(20, "cbl-upgrade-docs", db=source_db, channels=channels_sg)
+
+    time.sleep(2)
```
Comment (sridevi-15) [inline testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py:384]: no need of this sleep

**Review by sridevi-15 — CHANGES_REQUESTED** [review]

File: testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py (hunk 2 above)
Comment (mgcouchbase) [reply testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py:418]: done

File: testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py [hunk 5]
```diff
@@ -352,3 +355,67 @@ def _upgrade_db(args):
...
+    upgraded_cbl_doc_count = db.getCount(source_db)
+    cbl_doc_ids = db.getDocIds(source_db)
+    cbl_db_docs = db.getDocuments(source_db, cbl_doc_ids)
+    assert upgraded_cbl_doc_count == 20, " Number of docs count is not matching"
+    # Create few more docs and make sure the after upgrade we still create docs
+    db.create_bulk_docs(2, "cbl-upgrade-docs_new", db=source_db, channels=channels_sg)
+    upgraded_cbl_doc_count = db.getCount(source_db)
+    assert upgraded_cbl_doc_count > base_cbl_doc_count, " Number of docs count is not matching"
+    #
+    # log_info("*" * 20)
```
Comment (sridevi-15) [inline testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py]: As these lines are comment, can you remove them?

**Review by sridevi-15 — CHANGES_REQUESTED** [review]

**Review by sridevi-15 — APPROVED** [review]

**Pull request merged**